src/main/kotlin/org/rust/ide/core/overrideImplement/generateUtil.kt

```diff
@@ -0,0 +1,112 @@
+package org.rust.ide.core.overrideImplement
+
+import com.intellij.codeInsight.generation.ClassMember
+import com.intellij.codeInsight.generation.MemberChooserObject
+import com.intellij.codeInsight.generation.MemberChooserObjectBase
+import com.intellij.ide.util.MemberChooser
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.psi.PsiElement
+import com.intellij.ui.SimpleColoredComponent
+import com.intellij.ui.SimpleTextAttributes
+import io.netty.util.internal.chmv8.ConcurrentHashMapV8
+import org.rust.ide.icons.RsIcons
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsImplItem
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsTypeAlias
+import org.rust.lang.core.psi.impl.mixin.default
+import org.rust.lang.core.psi.impl.mixin.isAbstract
+import org.rust.lang.core.psi.util.childOfType
+import org.rust.lang.core.psi.util.trait
+import org.rust.utils.getFormattedParts
+import javax.swing.JTree
+
+sealed class FunctionOrType {
+    class Function(val function: RsFunction) : FunctionOrType()
+    class Type(val type: RsTypeAlias) : FunctionOrType()
+}
+
+private class RsFunctionChooserMember(val base: MemberChooserObjectBase, val member: FunctionOrType) : ClassMember {
+    private val text: String;
+
+    init {
+        when (member) {
+            is FunctionOrType.Function -> {
+                val (before, after) = member.function.getFormattedParts()
+                text = "${member.function.name}$after"
+            }
+            is FunctionOrType.Type -> {
+                text = "${member.type.name}"
+            }
+        }
+    }
+
+    override fun renderTreeNode(component: SimpleColoredComponent?, tree: JTree?) {
+        when (member) {
+            is FunctionOrType.Function -> {
+                component?.append("fn ", SimpleTextAttributes.REGULAR_BOLD_ATTRIBUTES)
+                component?.icon = RsIcons.FUNCTION
+            }
+            is FunctionOrType.Type -> {
+                component?.append("type ", SimpleTextAttributes.REGULAR_BOLD_ATTRIBUTES)
```

Comment:
I see that Kotlin doesn't add `fun` to functions in the same window (neither do Java or CLion, but they don't have such keywords). Maybe we don't need `fn` and `type` labels here because can simply understand the type of the item by the icon and signature. What do you think?